### PR TITLE
feat: DAH-2811 How to Apply page - Not Yet Open components

### PR DESCRIPTION
**Pull request by jimlin-sfgov**
## Description

Conditionally show two UI components, based on `listing.Application_Start_Date_Time`. Added fetching of listing data on page load.

## Jira ticket

https://sfgovdt.jira.com/browse/DAH-2811

### Version Control

- [x] branch name begins with `angular` if it contains updates to Angular code
- [x] branch name contains the Jira ticket number
- [x] PR name follows `type: TICKET-NUMBER Description` format, e.g. `feat: DAH-123 New Feature`

### Code quality

- [x] [the set of changes is small](https://google.github.io/eng-practices/review/developer/small-cls.html#what-is-small)
- [x] all automated code checks pass (linting, tests, coverage, etc.)
- [x] code irrelevant to the ticket is not modified e.g. changing indentation due to automated formatting
- [x] if the code changes the UI, it matches the UI design exactly

### Review instructions

- [x] instructions specify which environment(s) it applies to
- [x] instructions work for PA testers
- [x] instructions have already been performed at least once

### Request review

- [x] PR has `needs review` label
- [x] Use `Housing Eng` group to automatically assign reviewers, and/or assign specific engineers
- [x] If time sensitive, notify engineers in Slack

### CODE REVIEW

File: app/javascript/pages/howToApply/how-to-apply.tsx [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
-import { getSfGovUrl, renderInlineMarkup } from "../../util/languageUtil"
+import { getListing } from "../../api/listingApiService"
+
+import "./how-to-apply.scss"
 
 interface HowToApplyProps {
   assetPaths: unknown
 }
 
+const applicationsNotYetOpen = (listing: RailsSaleListing) =>
```
Comment (tallulahkay) [inline app/javascript/pages/howToApply/how-to-apply.tsx:23]: There is a listing state function now: https://github.com/SFDigitalServices/sf-dahlia-web/blob/main/app/javascript/util/listingUtil.ts#L41

File: app/javascript/pages/howToApply/how-to-apply.tsx [hunk 2]
```diff
@@ -1,14 +1,28 @@
-import React, { ReactNode } from "react"
-
+import React, { ReactNode, useEffect, useState, useContext } from "react"
```
Comment (jimlin-sfgov) [inline app/javascript/pages/howToApply/how-to-apply.tsx:1]: This file is now over the recommended max length of 250 lines of code. I'll ignore this issue in this PR, but I'm going to make a follow-up ticket to refactor by moving most of the components into separate files in `app/javascript/modules/howToApply`.

File: app/javascript/modules/listingDetailsAside/ListingDetailsApplicationDate.tsx [hunk 3]
```diff
@@ -13,14 +13,6 @@ export interface ListingDetailsApplicationDateProps {
...
 }
 
-export const formatTime = (time: string) => {
-  const formattedTime = dayjs(time).format("h:mm")
-  const hour = Number(dayjs(time).format("H"))
-  // \u00A0 is a non-breaking space
-  const suffix = hour >= 12 ? "\u00A0PM" : "\u00A0AM"
-  return `${formattedTime}${suffix}`
-}
-
```
Comment (jimlin-sfgov) [inline app/javascript/modules/listingDetailsAside/ListingDetailsApplicationDate.tsx:23]: extracting reusable code to keep things DRY

Comment (tallulahkay) [reply app/javascript/modules/listingDetailsAside/ListingDetailsApplicationDate.tsx:23]: 🏜️

File: app/assets/json/translations/react/en.json [hunk 4]
```diff
@@ -611,9 +611,11 @@
   "howToApplyPage.howToApplySection.step5.p1": "When you are ready, submit your application in 3 steps:",
   "howToApplyPage.howToApplySection.step5.p2": "The time and date you apply determines your place in line.",
   "howToApplyPage.howToApplySection.step5.title": "Submit your application and get your place in line",
+  "howToApplyPage.weAreNotAcceptingApplicationsYet": "We are not accepting applications yet.",
```
Comment (tallulahkay) [inline app/assets/json/translations/react/en.json:614]: Please run grunt translations

File: app/javascript/pages/howToApply/how-to-apply.tsx [hunk 5]
```diff
@@ -74,7 +109,7 @@ const BeforeYouStartSection = () => {
       <div>{t("howToApplyPage.beforeYouStartSection.eligibilityList.title")}</div>
       <ul className="mb-0 pt-2">
         {eligibilityListItems.map((item) => (
-          <li>
+          <li key={item.index}>
```
Comment (tallulahkay) [inline app/javascript/pages/howToApply/how-to-apply.tsx:112]: ty!

File: app/javascript/util/listingUtil.ts [hunk 6]
```diff
@@ -46,13 +46,13 @@ export const getFcfsSalesListingState = (listing: RailsSaleListing): ListingStat
     return ListingState.Closed
   }
 
-  if (new Date(listing.Application_Start_Date_Time) < new Date()) {
-    console.log("open")
+  if (dayjs(listing.Application_Start_Date_Time) < dayjs()) {
```
Comment (tallulahkay) [inline app/javascript/util/listingUtil.ts:49]: why this change?

File: app/javascript/pages/howToApply/how-to-apply.tsx [hunk 7]
```diff
@@ -186,14 +223,25 @@ const SubmitApplicationStep = () => {
         <li className="text-base">{t("howToApplyPage.howToApplySection.step5.listItem3")}</li>
       </ul>
       <div className="text-base">{t("howToApplyPage.howToApplySection.step5.p2")}</div>
+      {applicationsNotYetOpen(listing) && (
+        <div className="text-base pt-2">
```
Comment (tallulahkay) [inline app/javascript/pages/howToApply/how-to-apply.tsx:227]: I think this needs more padding on top - it's 24px in the design

File: app/javascript/pages/howToApply/how-to-apply.tsx [hunk 8]
```diff
@@ -228,22 +276,44 @@ const WhatHappensNextSection = () => {
...
-              <BeforeYouStartSection />
-              <HowToApplySection />
-              <WhatHappensNextSection />
+    <LoadingOverlay isLoading={!listing}>
+      <Layout title={t("pageTitle.howToApply")}>
+        <section className="flex md:px-5">
+          <article className="markdown max-w-5xl m-auto">
+            <div className="pt-4 md:py-0 max-w-3xl">
+              <div className="my-6 md:my-12 px-5">
+                {listing && (
```
Comment (tallulahkay) [inline app/javascript/pages/howToApply/how-to-apply.tsx:301]: suggestion: instead of checking for the listing here, change the two instances of `listing.Application_Start_Date_Time` to `listing?.Application_Start_Date_Time` - simplifies the code

**Review by tallulahkay — CHANGES_REQUESTED** [review]
Looks really good! Just a couple of minor comments!

File: app/javascript/util/listingUtil.ts (hunk 6 above)
Comment (jimlin-sfgov) [reply app/javascript/util/listingUtil.ts:49]: 1. We shouldn't log to the console
2. `dayjs` is what we use for _most_ date/time processing, for consistency I think we gradually move away from `Date` when possible

File: app/javascript/pages/howToApply/how-to-apply.tsx (hunk 8 above)
Comment (jimlin-sfgov) [reply app/javascript/pages/howToApply/how-to-apply.tsx:301]: Some counterpoints:
1. additional changes would also be needed to handle falsey params for `localizedFormat` and `formatTimeOfDay`
2. future components that use `listing` may also need to handle falsey values, I think it's more efficient to block rendering of all components until `listing` is loaded, like how it's done on the Listing Details page.

Let me know if those make sense.

Comment (jimlin-sfgov) [reply app/javascript/pages/howToApply/how-to-apply.tsx:301]: Update: I like the suggestion, if only because the markup looks a little cleaner. Authors of future components should feel free to revert if they need to.

Comment (tallulahkay) [reply app/javascript/pages/howToApply/how-to-apply.tsx:301]: Yeah I was going for the cleaner markup, but it is what it is

File: app/javascript/util/listingUtil.ts (hunk 6 above)
Comment (tallulahkay) [reply app/javascript/util/listingUtil.ts:49]: gotcha

**Review by tallulahkay — APPROVED** [review]
Looks good!
